gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -144,28 +142,18 @@
   private final boolean pkChunkingSkipCountCheck;
   private final boolean bulkApiUseQueryAll;
 
+  private WorkUnitState workUnitState;
+
   public SalesforceExtractor(WorkUnitState state) {
     super(state);
-    this.sfConnector = (SalesforceConnector) this.connector;
-
-    // don't allow pk chunking if max partitions too high or have user specified partitions
-    if (state.getPropAsBoolean(Partitioner.HAS_USER_SPECIFIED_PARTITIONS, false)
```

Comment:
How come this check was removed? If the user has specified partitions then that should override PK chunking.

Reply:
This part code was only for **PK chunking**. We don't need 2nd level PK-chunking any more. (if we have normal pre-partition, we should not use pk chunking. It burns out request quota)

I create separate function - getQueryResultIdsPkChunking for pkchunking. and made it not to depends on class member this.pkChunking, therefore removed pkChunking.